checkST2110.js

```diff
@@ -28,8 +28,8 @@ const groupPattern = /a=group:DUP\s+(\S+)\s+(\S+)/;
 const ssrcPattern = /a=ssrc:(\d+)\s/;
 const videoPattern = /video\s+(\d+)(\/\d+)?\s+(RTP\/S?AVP)\s+(\d+)/;
 const rtpmapPattern = /a=rtpmap:(\d+)\s(\S+)\/(\d+)\s*/;
-const fmtpPattern = /a=fmtp:(\d+)\s+(?:([^\s=;]+)(?:=([^\s;]+))?;\s+)*$/;
-const fmtpParams = /([^\s=;]+(?:=[^\s;]+)?);/g;
+const fmtpPattern = /a=fmtp:(\d+)\s+(?:([^\s=;]+)(?:=([^\s;]+))*(?:;\s*){0,1})+$/;
```

Comment:
Should the `fmtpPattern` allow `a=fmtp:42 a=b=c=`? (It does right now.)

Should it allow `a=fmtp:42` alone, or `a=fmtp:42 `, with a trailing space? (It doesn't at the moment.)

Reply:
The latter is intentional, the former is not.

Reply:
`a=fmtp:(\d+)\s+(?:([^\s=;]+)(?:=([^\s=;]+))?(?:;\s*){0,1})+$`
Maybe?

Reply:
I can't immediately find a reference which suggests it's invalid, although I certainly haven't seen it used like that before. Unfortunately the suggested fix doesn't appear to solve it. I'm having a play with alternatives, but perhaps if there's nothing to say it's invalid that's unnecessary.

Reply:
Hm, works for me at https://regex101.com/

But yes, if a value containing `=` isn't invalid, so be it. Still would make sense to change the `*` to `?` though?

Reply:
Conclusion via Slack chat was that the required regex is:
`a=fmtp:(\d+)\s+(?:([^\s=;]+)(?:=([^\s;]+))?)((?:;\s*)(?:([^\s=;]+)(?:=([^\s;]+))?))*(?:;\s*)?$`

This is made up of (in pseudo-code):

`fmtpElement` = `([^\s=;]+)(?:=([^\s;]+))?`

`fmtpSeparator` = `(?:;\s*)`

`fmtpPattern` = `a=fmtp:(\d+)\s+` `fmtpElement` + `(` `fmtpSeparator` + `fmtpElement` `)*` + `fmtpSeparator` `?$`

(I'm not sure whether you can build up regexes with string-like concatenation in Javascript?)
Obviously then the `fmtpParams` regex can also be built out of `fmtpElement` and `fmtpSeparator` too.

Reply:
I believe this is now reflected in the next commit.